pkg/api/metrics.go

```diff
@@ -37,7 +37,7 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext) response.Response {
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
 
-	resp, err := hs.queryDataService.QueryData(c.Req.Context(), c.SignedInUser, c.SkipCache, reqDTO, true)
+	resp, err := hs.queryDataService.QueryData(context.WithValue(c.Req.Context(), "logzioHeaders", c.Req.Header), c.SignedInUser, c.SkipCache, reqDTO, true) // LOGZ.IO CHANGE :: DEV-33325 Open expressions for Grafana 8.5.1
```

Comment:
nit: can make `logzioHeaders` a constant so it can be used in both adding to context and retrieving from the context